src/Drips.sol

```diff
@@ -385,17 +387,17 @@ library Drips {
     /// @return balance The current drips balance.
     function _currBalance(
         uint128 lastBalance,
-        uint64 lastUpdate,
-        uint64 defaultEnd,
+        uint32 lastUpdate,
+        uint32 defaultEnd,
         DripsReceiver[] memory receivers
     ) private view returns (uint128 balance) {
         balance = lastBalance;
         for (uint256 i = 0; i < receivers.length; i++) {
             DripsReceiver memory receiver = receivers[i];
-            uint64 start = receiver.start;
+            uint32 start = receiver.start;
             if (start == 0) start = lastUpdate;
-            uint64 end = defaultEnd;
-            if (receiver.duration != 0) end = _capTimestamp(uint72(start) + receiver.duration);
+            uint32 end = defaultEnd;
+            if (receiver.duration != 0) end = _capTimestamp(uint40(start) + receiver.duration);
```

Comment:
We basically adding to two uint32, so we need more space and `uint40` is the next in line.

This casting with different uint sizes, so many things to consider.  Need to check if there are any nice libraries doing these things correctly.